monai/apps/auto3d/data_analyzer.py

```diff
@@ -0,0 +1,541 @@
+        start = time.time()
+        ndas = self.data["image"]
+        ndas = [ndas[i] for i in range(ndas.shape[0])]
+        if "nda_croppeds" not in self.data.keys():
+            self.data["nda_croppeds"] = [self._get_foreground_image(_) for _ in ndas]
+        nda_croppeds = self.data["nda_croppeds"]
+        # perform calculation
+        case_stats = {
+            "image_stats": {
+                "shape": [list(_.shape) for _ in ndas],
+                "channels": len(ndas),
+                "cropped_shape": [list(_.shape) for _ in nda_croppeds],
+                # 'spacing': np.tile(np.array(self.data['image_meta_dict']['pixdim'][1:4]), [len(ndas),1]).tolist(),
+                "spacing": np.tile(np.diag(self.data["image_meta_dict"]["affine"])[:3], [len(ndas), 1]).tolist(),
+                "intensity": [self._stats_opt(_) for _ in nda_croppeds],
+            }
+        }
+        logger.debug(f"Get image stats spent {time.time()-start}")
+        return case_stats
+
+    def _get_case_image_stats_summary(self):
+        ###############################################################
+        """Update gather_summary"""
+        # this dictionary describes how to gather values in the summary
+        case_stats_summary = {
+            "image_stats": {
+                "shape": partial(self._stats_opt_summary, average=self.SUMMERY_AVERAGE),
+                "channels": self._stats_opt_summary,
+                "cropped_shape": partial(self._stats_opt_summary, average=self.SUMMERY_AVERAGE),
+                "spacing": partial(self._stats_opt_summary, average=self.SUMMERY_AVERAGE),
+                "intensity": partial(self._intensity_summary, average=self.SUMMERY_AVERAGE),
+            }
+        }
+        self.gather_summary.update(case_stats_summary)
+
+    def _get_case_foreground_image_stats(self):
+        ###############################################################
+        """Generate case_stats"""
+        # retrieve transformed data from self.data
+        start = time.time()
+        ndas = self.data["image"]
+        ndas = [ndas[i] for i in range(ndas.shape[0])]
+        ndas_l = self.data["label"]
+        if "nda_foreground" not in self.data.keys():
+            self.data["nda_foreground"] = [self._get_foreground_label(_, ndas_l) for _ in ndas]
+        nda_foreground = self.data["nda_foreground"]
+
+        case_stats = {"image_foreground_stats": {"intensity": [self._stats_opt(_) for _ in nda_foreground]}}
+        logger.debug(f"Get foreground image data stats spent {time.time()-start}")
+        return case_stats
+
+    def _get_case_foreground_image_stats_summary(self):
+        case_stats_summary = {
+            "image_foreground_stats": {"intensity": partial(self._intensity_summary, average=self.SUMMERY_AVERAGE)}
+        }
+        self.gather_summary.update(case_stats_summary)
+
+    @staticmethod
+    def label_union(x):
+        return list(set.union(*[set(np.array(_).tolist()) for _ in x]))
```

Comment:
Please add doc-string for this function and move it to the misc utils:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/utils/misc.py

Thanks.

Reply:
Fixed in a1055d8